ports/unix/unix_mphal.c

```diff
@@ -160,6 +171,10 @@ int mp_hal_stdin_rx_chr(void) {
     } else {
         main_term:;
 #endif
+        while (!input_available()) {
+            mp_handle_pending();
+            usleep(1);
```

Comment:
The problems @dpgeorge has mentioned would effectively disappear if we slept for 1 or 10 milliseconds instead of 1 microsecond.

Reply:
@embeddedt I would first like to try his suggestion executing uPy asynchronously without calling `mp_sched_schedule`.